catalog/core/catalog-core-attributeregistry/src/main/java/ddf/catalog/data/impl/AttributeRegistryImpl.java

```diff
@@ -33,15 +33,13 @@
     public AttributeRegistryImpl() {
         BasicTypes.BASIC_METACARD.getAttributeDescriptors()
                 .stream()
-                .forEach(this::registerAttribute);
+                .forEach(this::register);
     }
 
     @Override
-    public boolean registerAttribute(final AttributeDescriptor attributeDescriptor) {
-        Preconditions.checkArgument(attributeDescriptor != null,
-                "The attribute descriptor cannot be null.");
-        Preconditions.checkArgument(attributeDescriptor.getName() != null,
-                "The attribute name cannot be null.");
+    public boolean register(final AttributeDescriptor attributeDescriptor) {
```

Comment:
It looks like this attribute registry does not allow duplicate attributes but it is currently possible to have duplicate attributes for different metacard types.  This is a problem for the geojson and xml input transformers that you switched over to use this registry.